github/code-scanning.go

```diff
@@ -119,6 +119,38 @@ type AlertListOptions struct {
 	ListOptions
 }
 
+// AnalysesListOptions specifies optional parameters to the CodeScanningService.ListAnalysesForRepo method.
+type AnalysesListOptions struct {
+	// Return code scanning analyses belonging to the same SARIF upload.
+	SarifID string `url:"sarif_id,omitempty"`
+
+	// Return code scanning analyses for a specific branch reference. The ref can be formatted as refs/heads/<branch name> or simply <branch name>.
+	Ref string `url:"ref,omitempty"`
+
+	ListOptions
+}
+
+// Analysis represents an individual GitHub Code Scanning Analysis on a single repository.
+//
+// GitHub API docs: https://docs.github.com/en/rest/reference/code-scanning#list-code-scanning-analyses-for-a-repository
+type Analysis struct {
```

Comment:
Can this be merged with the existing `SarifAnalysis` struct?
If not, can we give it a more specific name?

Reply:
It can't because `SarifAnalysis` is used for POST body params while `Analysis` is used to unmarshal json response of listing code scanning analysis instances. 

I thought about renaming to to `ScanningAnalysis` or `CodeScanningAnalysis` but I felt that was redundant with the service name. `CodeScanningService`. What are your thoughts on this?

Reply:
The problem is that this name is global throughout the `package github` namespace. So even though it is _used_ within the `CodeScanningService`, it is really `github.Analysis` which is pretty general, so if you don't mind switching to `ScanningAnalysis`, I think that would be preferable.  Thank you!

Reply:
Oh yes you're right. I updated it. Thanks for the review!